src/prefect/engine.py

```diff
@@ -1466,6 +1480,23 @@ async def orchestrate_task_run(
                 )
                 args, kwargs = parameters_to_args_kwargs(task.fn, resolved_parameters)
 
+                # update task run name
+                if not run_name_set and task.task_run_name:
+                    sig = inspect.signature(task.fn)
+                    for kwarg, val in sig.parameters.items():
+                        if val.default is not val.empty:
+                            resolved_parameters.setdefault(kwarg, val.default)
+                    task_run_name = task.task_run_name.format(**resolved_parameters)
+                    await client.set_task_run_name(
+                        task_run_id=task_run.id, name=task_run_name
+                    )
```

Comment:
We may want to skip this API call if the name is the same as `task_run.name` for performance

Reply:
I don't think I'm following you here - without this call the name wouldn't get updated?

Reply:
If, after formatting, the name has not changed we should not make the API call.

Reply:
That seems like a really specific edge case that isn't worth coding to avoid - a user would have to do something like `task_run_name="task-name-0"`

Reply:
Oh sorry I thought we were using the user-configured name during creation with placeholders for parameters that are futures. We might want to use the configured name during creation to avoid confusion especially for users that aren't using futures. I think we can tackle that as an enhancement after the mvp though.